api-notification.go

```diff
@@ -156,6 +157,14 @@ func (c Client) ListenBucketNotification(bucketName string, accountArn Arn, done
 			return
 		}
 
+		// Check if we are calling Minio server which supports listening to bucket notifications
+		if !strings.HasPrefix(resp.Header.Get("Server"), "Minio/") {
```

Comment:
According to this https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.38, proxies are not allowed to modify the server Header